src/widget/city_building_ghost.c

```diff
@@ -419,6 +429,27 @@ static void set_roamer_path(building_type type, int size, const map_tile *tile,
     }
 }
 
+static void draw_mausoleum_desirability_range(const map_tile *tile, building_type type, int building_size)
+{
+    const model_building *model = model_get_building(type);
+    int positive_range = model->desirability_range;
+
+    int desirability_value = model->desirability_value;
+    int desirability_step_size = model->desirability_step_size;
+    int negative_range = 0;
+
+    while (desirability_value < 0) {
+        desirability_value += desirability_step_size;
+        negative_range++;
+    }
+
+    city_view_foreach_tile_in_range(tile->grid_offset, building_size, positive_range, city_building_ghost_draw_bonus_range);
+
+    if (type != BUILDING_NYMPHAEUM) {
+        city_view_foreach_tile_in_range(tile->grid_offset, building_size, negative_range, city_building_ghost_draw_malus_range);
```

Comment:
```suggestion
        city_view_foreach_tile_in_range(tile->grid_offset, building_size + positive range, negative_range - positive range, city_building_ghost_draw_malus_range);
```

I believe you can make this a little bit more performant by leaving a "hole" to be filled with just the positive range. This needs testing to confirm it works though.

Reply:
it doesn't work

Reply:
Ah, shame!